content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1389 @@
+
+[source]
+----
+sudo sh -c "iptables-save > /etc/iptables.rules"
+----
+
+The iptables-restore command will need to be executed manually, or your
+system configured to automatically run it on boot, against the
+/etc/iptables.rules file you have created, in order for your iptables
+configuration to be retained across reboots. 
+On Ubuntu fastest way is to install iptables-persistent after configuring iptables - 
+it will automatically create necessery files from current configuration and load
+them on boot.
+
+[source]
+----
+sudo apt-get install iptables-persistent
+----
+
+See https://help.ubuntu.com/community/IptablesHowTo for other Ubuntu
+options. 
+There are many other resources describing this; please consult
+your system's documentation or search on the internet for information
+specific to your flavor of Linux.
+
+If you are unsure at all about what kind of system you have, consult
+that system's documentation on how to update iptables configuration.
+
+==== Alternate Solution
+
+Note that an earlier version of this documentation had different
+iptables commands for forwarding ports. 
+I found that these did not work for me, and so I researched and 
+came up with the information above. 
+I do not believe that DNAT is required, as it is typically used to publish a
+service from an internal network to a publicly accessible IP. 
+But, if the above forwarding commands do not work for you, you might want to
+consider trying the commands below. 
+Keep in mind that the rules to allow incoming traffic are still required in 
+order for these rules to work:
+
+*iptables -t nat -A PREROUTING -i eth0 -p tcp --dport 80 -j DNAT
+--to-destination 127.0.0.1:8080*
+
+The first rule uses -j DNAT target of the PREROUTING chain in NAT to
+specify a destination IP address and port where incoming packets
+requesting a connection to your Jenkins service can be forwarded.
+
+If you have a default policy of DROP in your FORWARD chain, you will
+need to add a rule to forward all incoming traffic.
+
+*iptables -A FORWARD -i eth0 -m state --state NEW -m tcp -p tcp -d
+127.0.0.1 --dport 8080 -j ACCEPT*
+
+For HTTPS you will need to repeat the above steps but specify port 443
+instead of port 80.
+
+==== Using firewalld
+
+Newer Linux distributions (CentOS 7, RHEL 7, etc.) ship with firewalld
+which serves as a front-end for iptables.  
+Configuration thru firewalld is done via the *firewall-cmd* command.  
+Instead of using any of the iptables commands mentioned above, 
+all you should need to do is something like:
+
+[source]
+----
+sudo firewall-cmd --add-port=80/tcp --permanent   (allow incoming connections on port 80.  You can also use --add-service=http instead of adding a port number)
+sudo firewall-cmd --add-port=443/tcp --permanent  (allow incoming connections on port 443.  You can also use --add-service=https instead of adding a port number)
+sudo firewall-cmd --add-forward-port=port=80:proto=tcp:toaddr=127.0.0.1:toport=8080 --permanent
```

Comment:
Place comments about the commands and group the operations by port.
```suggestion
# allow incoming connections on port 80.
# You can also use --add-service=http instead of adding a port number
sudo firewall-cmd --add-port=80/tcp --permanent
sudo firewall-cmd --add-forward-port=port=80:proto=tcp:toaddr=127.0.0.1:toport=8080 --permanent

# allow incoming connections on port 443.
# You can also use --add-service=https instead of adding a port number
sudo firewall-cmd --add-port=443/tcp --permanen
t
```